advanced_samples/CSV/dotnet/CSVtoOCS/Program.cs

```diff
@@ -0,0 +1,303 @@
+﻿
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using OSIsoft.Data;
+using OSIsoft.Data.Reflection;
+using OSIsoft.Identity;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Globalization;
+using CsvHelper;
+using IdentityModel.Client;
+using IdentityModel.OidcClient;
+
+namespace CSVtoOCS
+{
+    public class Program
+    {
+        public static Exception toThrow = null;
+        public static bool success = true;
+
+        static List<TemperatureReadingsWithIds> dataList;
+        static IEnumerable<string> streamsIdsToSendTo;
+        static ISdsDataService dataService;
+        static ISdsMetadataService metaService;
+        private static bool createStreams = true;
+        static string stream1ID = "stream1", stream2ID = "stream2", typeID = "TemperatureReadings";
+        static bool test = false;
+
+        static void Main(string[] args)
+        {
+            string fileLocationIn = "datafile.csv";
+            if (args.Length > 0)
+            {
+                if(args[0] != "%LAUNCHER_ARGS%")
+                    fileLocationIn = args[0];
+            }
+            MainAsync(fileLocation: fileLocationIn).GetAwaiter().GetResult();
+        }
+
+        public static async Task<bool> MainAsync(bool testIn = false, string fileLocation = "datafile.csv")
+        {
+            test = testIn;
+            success = true;
+
+            try
+            {
+                // Import data in.  Use csv reader and custom class to make it simple
+                using (var reader = new StreamReader(fileLocation))
+                using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
+                {
+                    dataList = csv.GetRecords<TemperatureReadingsWithIds>().ToList();
+                }
+                // Use Linq to get the distinct StreamIds we need.  
+                streamsIdsToSendTo = dataList.Select(dataeEntry => dataeEntry.StreamId).Distinct();
+
+                //Get Configuration information about where this is sending to
+                IConfiguration configuration = new ConfigurationBuilder()
```

Comment:
Polaris doesn't like this since configuration is never disposed. I've been moving these to a static variable instead and it's happier.